tempodb/search/pipeline.go

```diff
@@ -91,6 +87,52 @@ func NewSearchPipeline(req *tempopb.SearchRequest) Pipeline {
 	return p
 }
 
+// rewriteTagLookup intercepts certain tag/value lookups and rewrites them. It returns
+// true if the tag lookup should be excluded from the remaining tag/value lookups because
+// the it was rewritten into a different filter altogether.  Otherwise it returns false,
+// and a new set of tag/value strings to use, which will either be the original inputs
+// or rewritten lookups.
+func (p *Pipeline) rewriteTagLookup(k, v string) (skip bool, newk, newv string) {
+	switch k {
+	case SecretExhaustiveSearchTag:
+		// Perform an exhaustive search by:
+		// * no block or page filters means all blocks and pages match
+		// * substitute this trace filter instead rejects everything. therefore it never
+		//   quits early due to enough results
+		p.tracefilters = append(p.tracefilters, func(s tempofb.Trace) bool {
+			return false
+		})
+		// Skip
+		return true, "", ""
+
+	case ErrorTag:
+		// Convert error=true|false into status.code=2|1
+		b, err := strconv.ParseBool(v)
+		if err != nil {
+			// Not a bool string = fall-through
+			break
+		}
+
+		if b {
+			// Error = true
+			return false, StatusCodeTag, strconv.Itoa(int(v1.Status_STATUS_CODE_ERROR))
+		}
+		// Else fall-through
```

Comment:
I think we can simplify this a little bit:
```go
case ErrorTag:
	// Convert error=true into status.code=2
	if v == "true" {
		return false, StatusCodeTag, strconv.Itoa(int(v1.Status_STATUS_CODE_ERROR))
	}
	// Else fall-through
```

This will be a bit more restrictive because `strconv.ParseBool` also accepts `t` and `1` as `true`, but I don't think that's a bad thing. WDYT?